native/NetworkProxy/NetworkProxy.cpp

```diff
@@ -0,0 +1,321 @@
+﻿// NetworkProxy.cpp : Defines the entry point for the application.
+//
+
+#include "stdafx.h"
+#include "Strsafe.h"
+#include "NetworkProxy.h"
+#include "NetworkMonitor.h";
+#include <functional>
+#include <memory>
+#include <string>
+
+using namespace std;
+using namespace NetworkProxyLibrary;
+using namespace Windows::Data;
+
+
+/*************************************************************/
+#define WM_COPYDATA 0x004A
+#define WM_PROCESSCOPYDATA WM_USER + 1
+
+#define MAX_LOADSTRING 100
+
+// Global Variables:
+HINSTANCE hInst;                                // current instance
+WCHAR szTitle[MAX_LOADSTRING];                  // The title bar text
+WCHAR szWindowClass[MAX_LOADSTRING];            // the main window class name
+
+NetworkMonitor* m_networkMonitor;
+HWND m_serverHwnd;
+DWORD m_edgeProcessId;
+HWND m_hMainWnd;
+
+
+// Forward declarations of functions included in this code module:
+ATOM                MyRegisterClass(HINSTANCE hInstance);
+BOOL                InitInstance(HINSTANCE, int);
+LRESULT CALLBACK    WndProc(HWND, UINT, WPARAM, LPARAM);
+void OnMessageReceived(const wchar_t* message);
+void SendMessageToWebSocket(_In_ const wchar_t* message);
+
+int APIENTRY wWinMain(_In_ HINSTANCE hInstance,
+                     _In_opt_ HINSTANCE hPrevInstance,
+                     _In_ LPWSTR    lpCmdLine,
+                     _In_ int       nCmdShow)
+{
+    UNREFERENCED_PARAMETER(hPrevInstance);
+    UNREFERENCED_PARAMETER(lpCmdLine);
+
+    // TODO: Place code here.
+    const wstring PIdParam = L"--process-id=";        
+    wstring commandLine = GetCommandLine();
+    auto startPIdParam = commandLine.find(PIdParam.c_str());
+
+    const wstring launchParam = L"--startlistener";
+    bool isAutoLaunchActive = commandLine.find(launchParam.c_str()) != string::npos;
+
+    if (startPIdParam == string::npos)
+    {       
+        throw ref new InvalidArgumentException(L"Required argument to start the application: --process-id=%processId%");
+    }
+        
+    auto endPIdParam = commandLine.find(L"-", startPIdParam + PIdParam.length());
+    if (endPIdParam == string::npos)
+    {
+        wstring paramValue = commandLine.substr(startPIdParam + PIdParam.length());
+        m_edgeProcessId = _wtol(paramValue.c_str());
```

Comment:
I am not sure to understand well the suggestion. The pid is validated in the NetworkMonitor::StartListeningEdgeProcess. If the pid is not found in the processes list an exception is thrown. What could I validate there, if the number stored in positive for instance?